gdal/port/cpl_multiproc.cpp

```diff
@@ -292,13 +292,22 @@ CPLMutex*& CPLCreateOrAcquireMasterMutexInternal(double dfWaitInSeconds = 1000.0
     // with static storage duration is thread-safe in C++11
     static CPLMutex *hCOAMutex = CPLCreateUnacquiredMutex();
 
-    // Erroneous call to CPLCleanupMasterMutex or memory issues
-    CPLAssert( hCOAMutex );
+    // WARNING: although adding an CPLAssert(hCOAMutex); might seem logical
+    // here, do not enable it (see comment below). It calls CPLError that
+    // uses the hCOAMutex itself leading to recursive mutex acquisition
+    // and likely a stack overflow.
 
     if ( !hCOAMutex )
     {
-        // In case of unexpected call to CPLCleanupMasterMutex,
-        // attempt this NOT thread-safe re-construction of the master mutex.
+        // Fall back to this, ironically, NOT thread-safe re-initialisation of
+        // hCOAMutex in case of a memory error or call to CPLCleanupMasterMutex
+        // sequenced in an unusual, unexected or erroneous way.
```

Comment:
```suggestion
        // sequenced in an unusual, unexpected or erroneous way.
```

Reply:
Thanks. Fixed.

p.s. Using the GitHub feature to "Commit suggestion" for the first time. Handy.